volatility3/framework/plugins/linux/lsblk.py

```diff
@@ -0,0 +1,447 @@
+
+        for kobj in class_kset.list.to_list(
+            vmlinux.symbol_table_name + constants.BANG + "kobject", "entry"
+        ):
+            subsys_private = vmlinux.object(
+                object_type="subsys_private",
+                offset=kobj.vol.offset - kobj_off - subsys_off,
+                absolute=True,
+            )
+            if subsys_private.member("class") == block_class.vol.offset:
+                break
+
+        klist_devices = subsys_private.klist_devices
+        for device in self._device_iterator(vmlinux, klist_devices):
+            # Before v5.11, partitions are represented by hd_struct instead of block_device
+            if not vmlinux.has_type("hd_struct"):
+                block_device = vmlinux.object(
+                    object_type="block_device",
+                    offset=device.vol.offset - block_device_offset,
+                    absolute=True,
+                )
+                gendisk = block_device.bd_disk
+
+                # lsblk by default skips over ram devices
+                try:
+                    if utility.array_to_string(gendisk.disk_name).startswith("ram"):
+                        continue
+                except exceptions.InvalidAddressException:
+                    continue
+            else:
+                hd_struct = vmlinux.object(
+                    object_type="hd_struct",
+                    offset=device.vol.offset - hd_struct_offset,
+                    absolute=True,
+                )
+                gendisk = vmlinux.object(
+                    object_type="gendisk",
+                    offset=hd_struct.vol.offset - gendisk_offset,
+                    absolute=True,
+                )
+                try:
+                    if utility.array_to_string(gendisk.disk_name).startswith("ram"):
+                        continue
+                except exceptions.InvalidAddressException:
+                    continue
+
+                block_device = self._get_block_device(vmlinux, device.devt)
+                if not block_device:
+                    continue
+
+            try:
+                size = block_device.bd_inode.i_size
+            except exceptions.InvalidAddressException:
+                continue
+            # lsblk does not display devices with a size of 0
+            if not size:
+                continue
+
+            GENHD_FL_REMOVABLE = 1 << 0
+            removable = gendisk.flags & GENHD_FL_REMOVABLE
+
+            read_only = self._get_read_only(vmlinux, gendisk)
+
+            name = self._get_name(vmlinux, block_device, gendisk)
+
+            device_type = self._get_type(vmlinux, block_device, gendisk)
+
+            mountpoint = self._get_mountpoint(vmlinux, block_device)
+
+            yield 0, (
+                name,
+                self._major(device.devt),
+                self._minor(device.devt),
+                removable,
+                read_only,
+                self._format_bytes(size),
```

Comment:
Please don't format raw data that's returned from the plugin.  This might be being used to another plugin or some other code, so turning it into a string for a human to read creates *more* work for other plugins.  The UI should figure out how such things should be formatted.  If you want something like this it should inherit from `int` in the `format_hints` file, that way it only affects the CLI and a web user interface might find a better way of being able to display the abbreviated value but also let the user see the full value (via a tooltip for example).

I think I'm going to say this one *is* show stopper and the plugin won't be accepted until this has been changed.  This *must* return the full data as an integer so that other tools can use the value.  The format hint can be added later as part of a separate PR, or as part of this one.  Either would be acceptable.

Reply:
I added it to format_hints